autode/opt/optimisers/rfo.py

```diff
@@ -121,7 +122,9 @@ def _take_step_within_trust_radius(
             # Note because the transformation is not linear this will not
             # generate a step exactly max(∆x) ≡ α, but is empirically close
             factor = self.alpha / max_component
-            new_coords = self._coords + self.alpha / max_component * delta_s
+            step = factor * delta_s
 
-        self._coords = new_coords
+        self._coords.allow_unconverged_back_transform = False
+        self._coords = self._coords + step
+        self._coords.allow_unconverged_back_transform = True
```

Comment:
@t-young31 So basically, I wanted the new coordinates to be converged when you set `self._coords = something`. But as I said above, quite a lot of time, RFO takes a large step, so if the unconverged_back_transform is False by default, then before scaling down to trust radius, it will throw an exception and stop. But we don't want that. So, I made unconverged_back_transform True by default. That is okay, because the first RFO step is just a guess, and after scaling back to trust radius, it will probably converge.

Then just before the moment it stores the new coordinates, the flag is set to False. This ensures that the new coordinates are always converged. Then the flag has to be set to True again (as coords + step for DIC returns coords.copy() + step, so it inherits the flag from the previous coords). If it is not set to True, then in the next RFO step, it might throw an exception.

I don't know any other elegant way to handle this?

Reply:
gotcha  – thanks. Could we perhaps set `self._coords.allow_unconverged_back_transform = True` directly before where the unconverged `self._coords` is allowed?

Reply:
done!